### ISSUE TITLE
SFTP Open File crashes app

### ISSUE THREAD

**Issue by hyproman** [PhilippC/keepass2android#2263]
The app crashes after tapping Okay on the SFTP Open File dialog box.

**Version:** 1.09e-r3

**Steps to reproduce:**
1. From main/initial panel, tap Open File
2. Tap SFTP
3. Fill out connection details
4. Tap OK

**Expected:** The file chooser panel should be displayed
**Actual:** The main panel is displayed

logcat shows the app crashes due to target SDK compatibility issues:
```
D KP2A_FC : getLaunchFileChooserIntent
E keepass2android: Invalid ID 0x00000000.
...
D KP2A_FC_P: lastPathSegment:check_connection
D KP2A_FC_P: Check connection...
E ImeBackDispatcher: Ime callback not found. Ignoring unregisterReceivedCallback. callbackId: 9661337
E AndroidRuntime: FATAL EXCEPTION: Thread-4
E AndroidRuntime: Process: keepass2android.keepass2android, PID: 7547

E AndroidRuntime: java.lang.IllegalArgumentException: keepass2android.keepass2android: Targeting S+ (version 31 and above) requires that one of FLAG_IMMUTABLE or FLAG_MUTABLE be specified when creating a PendingIntent.
E AndroidRuntime: Strongly consider using FLAG_IMMUTABLE, only use FLAG_MUTABLE if some functionality depends on the PendingIntent being mutable, e.g. if it needs to be used with inline replies or bubbles.
E AndroidRuntime:        at android.app.PendingIntent.checkFlags(PendingIntent.java:404)
E AndroidRuntime:        at android.app.PendingIntent.getActivityAsUser(PendingIntent.java:487)
E AndroidRuntime:        at android.app.PendingIntent.getActivity(PendingIntent.java:473)
E AndroidRuntime:        at android.app.PendingIntent.getActivity(PendingIntent.java:437)
E AndroidRuntime:        at keepass2android.javafilestorage.SftpUserInfo$1.run(SftpUserInfo.java:54)
```

**Comment by hyproman** [comment]
Well I'm officially confused...

This is a duplicate of bug #2223, which I fixed in #2234. It was merged into the mainline well before **v1.09e-r3** was released. I just checked out the **v1.09e-r3** tag and the fix is there. I just built and installed the (Net) debug APK and confirmed it does not crash.

It would seem the **v1.09e-r3** git tag does not reflect the **v1.09e-r3** release APKs?

**Comment by PhilippC** [comment]
I checked this and it is indeed confusing. I found one mistake: I forgot to commit & push the modified Manifest files (increasing version code & number) and strings.xml (for the changelog) and then tagged the wrong commit (because I did not notice I had uncommited changes). I fixed this, but this does not explain what you saw.

In my local repository, your change is present, and the .aar files for JavaFileStorage have been built 2 hours before I built the final apk which makes me assume that the fix should correctly be present.

I will upload a new release shortly and will make sure to `make clean` before building.

**Comment by PhilippC** [comment]
I published 1.09e-r4 - can you check if it the crash is gone there?

**Comment by hyproman** [comment]
Sadly 1.09e-r4 has the same issue. The line number in the crash indicates that commit isn't included for some reason. I also checked out the 1.09e-r4 tag and the commit is included there. Weird.
```
03-09 15:47:17.124  2223  3344 E AndroidRuntime: FATAL EXCEPTION: Thread-7
03-09 15:47:17.124  2223  3344 E AndroidRuntime: Process: keepass2android.keepass2android, PID: 2223
03-09 15:47:17.124  2223  3344 E AndroidRuntime: java.lang.IllegalArgumentException: keepass2android.keepass2android: Targeting S+ (version 31 and above) requires that one of FLAG_IMMUTABLE or FLAG_MUTABLE be specified when creating a PendingIntent.
03-09 15:47:17.124  2223  3344 E AndroidRuntime: Strongly consider using FLAG_IMMUTABLE, only use FLAG_MUTABLE if some functionality depends on the PendingIntent being mutable, e.g. if it needs to be used with inline replies or bubbles.
03-09 15:47:17.124  2223  3344 E AndroidRuntime:        at android.app.PendingIntent.checkFlags(PendingIntent.java:404)
03-09 15:47:17.124  2223  3344 E AndroidRuntime:        at android.app.PendingIntent.getActivityAsUser(PendingIntent.java:487)
03-09 15:47:17.124  2223  3344 E AndroidRuntime:        at android.app.PendingIntent.getActivity(PendingIntent.java:473)
03-09 15:47:17.124  2223  3344 E AndroidRuntime:        at android.app.PendingIntent.getActivity(PendingIntent.java:437)
03-09 15:47:17.124  2223  3344 E AndroidRuntime:        at keepass2android.javafilestorage.SftpUserInfo$1.run(SftpUserInfo.java:54)
03-09 15:47:17.171  2223  2223 D KP2A    : onAR
```
![Screenshot_20230309-154619_Keepass2Android](https://user-images.githubusercontent.com/23566299/224156603-4a7222fc-2df2-4d0c-abff-78a35db2583d.png)

**Comment by PhilippC** [comment]
Hmm. Can you (privately) share an account for sftp so I can test this
myself? BTW you might try to install the build artifact from github and see
how this behaves.

Rick Brown ***@***.***> schrieb am Do., 9. März 2023, 22:00:

**Comment by hyproman** [comment]
@PhilippC I just emailed you SFTP test account details. When I have a few minutes I'll try out the build artifact and report back.

**Comment by hyproman** [comment]
The latest [artifact build](https://github.com/PhilippC/keepass2android/actions/runs/4378319671) for tag 1.09e-r4 (Net) **does** work.

However the (Net) apk on the v1.09e-r4 [Release page](https://github.com/PhilippC/keepass2android/releases/tag/v1.09e-r4) crashes

**Comment by PhilippC** [comment]
please verify this is fixed in https://github.com/PhilippC/keepass2android/releases/tag/v1.09e-r5. When I just wanted to test again, I wasn't able to reach your server , but in previous tests it seemed to be fine.

**Comment by hyproman** [comment]
I wish I had better news to report.

I just downloaded and clean installed **v1.09-r5** (net) from your link above and it still crashes trying to connect to the SFTP server. Exact same stack trace as before, including line numbers (that do not line up to `SftpUserInfo.java` in source). Actually did it three times to convince myself that I'm not crazy. Worth noting that the About panel does show **v1.09-r5**.

If I check out **v1.09e-r5** tag and clean build using `make Flavor=Net Configuration=Release` and install/use the signed Release APK, it does not crash. Similarly, clean building without `Configuration=Release` and installing/using the signed Debug APK does not crash.

It still feels to me like somehow the java bits aren't being cleanly built? That's the only way I can see that the version number would be up-to-date but the Java code be behind.

As for the test server, it looks like my ISP blocked your most-recent connection attempt, citing "IP Reputation"; I guess they think your address is suspicious. I don't have any control over what they are doing there (wish I did). I'll keep it up and running but also will look for a better alternative.
